src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGTabButton.cs

```diff
@@ -0,0 +1,189 @@
+﻿using TMPro;
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+namespace RegressionGames
+{
+    /**
+     * <summary>
+     * A button that is intended to be used with other buttons of the same type. These buttons should be used to toggle
+     * visible content or settings
+     * </summary>
+     */
+    public class RGTabButton : MonoBehaviour, IPointerEnterHandler, IPointerExitHandler, IPointerDownHandler
+    {
+        // is this button currently selected
+        public bool isActive;
+
+        public UnityEvent onClick;
+
+        private Image _background;
+
+        private TMP_Text _label;
+
+        private Color _fromColor;
+
+        private Color _toColor;
+
+        private float _colorChangeAlpha;
+
+        private bool _isChangingColor;
+
+        private static readonly Color DefaultColor = new (1, 1, 1, 0.01f);
+
+        private static readonly Color HoverColor = new (1, 1, 1, 0.05f);
+
+        private static readonly Color ActiveColor = Color.white;
+
+        /**
+         * <summary>
+         * Ensure that the required fields are set, and set the active state
+         * </summary>
+         */
+        public void Start()
+        {
+            var background = GetComponent<Image>();
+            if (background != null)
+            {
+                _background = background;
+            }
+            else
+            {
+                Debug.LogError("RGTabButton is missing its background");

```

Comment:
last comment on this, but we shouldn't need all these checks / log statements for misconfigured game objects